packages/conjure-client/src/fetchBridge/blobReadableStreamAdapter.ts

```diff
@@ -15,11 +15,13 @@
  * limitations under the License.
  */
 
-import "web-streams-polyfill";
+import { ReadableStream as PonyfilledReadableStream } from "web-streams-polyfill/ponyfill";
 
-export function blobToReadableStream(blobPromise: Promise<Blob>): ReadableStream<Uint8Array> {
-    return new ReadableStream({
-        start: controller => {
+export function blobToReadableStream(
+    blobPromise: Promise<Blob>,
+): ReadableStream<Uint8Array> | PonyfilledReadableStream<Uint8Array> {
```

Comment:
nit: Is it a bit strange for `PonyfilledReadableStream<T>` to be exposed as a return type of this function? Ideally the polyfill usage of this function is an implementation detail that doesn't affect its calling API contract?

Assuming this was added since `PonyfilledReadableStream<T>` isn't a subset of `ReadableStream<T>`?

Reply:
I do think it's strange, but unfortunately it is necessary.

```
Type 'ReadableStream<Uint8Array> | ReadableStream<...>' is not assignable to type 'ReadableStream<Uint8Array>'.
  Type 'import("/Volumes/git/conjure-typescript-runtime/node_modules/web-streams-polyfill/dist/types/polyfill").ReadableStream<Uint8Array>' is not assignable to type 'ReadableStream<Uint8Array>'.
    Types of property 'getReader' are incompatible.
      Type '{ ({ mode }: { mode: "byob"; }): ReadableStreamBYOBReader_2; (): ReadableStreamDefaultReader_2<Uint8Array>; }' is not assignable to type '() => ReadableStreamDefaultReader<Uint8Array>'.
```

Reply:
Gotcha. Thanks for copying out the compile error.

Although I dislike `as` assertions, I'm a bit conflicted on whether an `as ReadableStream<Uint8Array>` type assertion in this function might be appropriate. The `| PonyfilledReadableStream<Uint8Array>` addition to the return type here might mean downstream consumers have to cast that portion away later. If a cast somewhere in the call stack is necessary, it might be best to do it here where it's introduced?

What do you think?

Reply:
Filling in some offline discussion here:

The ponyfill typing and the native `ReadableStream` type don't have sufficient overlap to cast, so we have to cast as `unknown` first. Currently we cast as `any` here, downstream of `blobToReadableStream`: https://github.com/palantir/conjure-typescript-runtime/blob/70547a62d8d8a93f9fc33a5b629e3cc1167b3b82/packages/conjure-client/src/fetchBridge/fetchBridge.ts#L130

So this shouldn't affect downstream consumers regardless, but changing anyway to prevent any future change to that cast unintentionally affecting downstream consumers. Also, @gluxon pointed out we probably should have needed to change the return type here: https://github.com/palantir/conjure-typescript-runtime/blob/70547a62d8d8a93f9fc33a5b629e3cc1167b3b82/packages/conjure-client/src/fetchBridge/fetchBridge.ts#L215

Not sure why that wasn't causing a compilation error